### ISSUE TITLE
Serach by uid

### ISSUE THREAD

**Issue by grekpg** [barbushin/php-imap#367]
I'm looking how to search not received messages - 

for plain PHP I found : 

`$uids = imap_search ( $mbox, "SINCE \"$date\"", SE_UID );`

for this-class I use

` $emails = $mailbox->searchMailbox();`

so how to get uid for this search  - and reuse in next search. I found only by date - but date don't have time so I every time get same messages. 

i looking for somme like this 

` $emails = $mailbox->searchMailbox('UID ' . $latest_uid . ':*');`

$latest_uid = $mailbox-> ge last uid ? 

I found setting : 

` $mailbox->setImapSearchOption(SE_UID);`

but what now ? how to set / get uid ?

**Comment by Sebi94nbg** [comment]
Should look like this, when I understood you correctly:
```php
$emails = $mailbox->searchMailbox("SINCE '$date'");
```

`$emails` will then contain the SE_UIDs of all found emails. This class is using the SE_UIDs by default.

Check out this example: https://github.com/barbushin/php-imap/blob/master/examples/get_and_parse_unseen_emails.php

**Comment by grekpg** [comment]
Thank fo reply.

but 

`$emails = $mailbox->searchMailbox("SINCE '$date'");`

If I run every 5 minutes - I get today messages ? I read that is not date time but date. 

or in next check i should save SE_UID and put it ?

I just want get all new messages (not unseen in box - if user set read this email in other client i want get it but get only not seen for this app)

**Comment by Sebi94nbg** [comment]
I recommend to leave the defaults as they are. I see no reason to change this.

By default, the `php-imap` class is already using the `SE_UID`, so you don't need to define / configure this explicitly.

When I remember correct, the other IDs (`SE_FRE`) are slowing down the entire search result.

Your last sentence is kinda confusing. Which emails (messages) do you want to get? ALL (unseen and seen) emails from the current day? Or do you only want to get new (unseen) emails from today?

I recommend the following solution:

- Get all unseen (not read yet) emails (at all or only from today or from the last hour)
- Parse these emails and do with them, what you want to do
- Mark successful parsed emails as seen (read)
- Repeat these steps in a regular time period

So, you might want this:

```php
$emails = $mailbox->searchMailbox("UNSEEN SINCE '$date'");
```

Should look like this:
```php
$emails = $mailbox->searchMailbox("UNSEEN SINCE '8 August 2008'");
```

Instead of `UNSEEN`, you can also use `NEW`.

`UNSEEN` returns emails (messages), which are not marked as read yet, while `NEW` returns just new arrived messages - so it can be even a seen (read) email (message).

For further information check out the criteria here: https://www.php.net/manual/de/function.imap-search.php

**Comment by Sebi94nbg** [comment]
I'll close this now. Feel free to comment this or open a new issue.

**Comment by avinash403** [comment]
There are cases where you do not want to mark the messages are read but want to fetch mails which are not fetched already.
It would be nice to have a search option like `"UID1:UID2"`, where we could have given range of ids to be fetched. This already exists in `php`'s `imap_fetch_overview` but not in `imap_search`.

**Comment by avinash403** [comment]
Was just going through the code and found this line
```
        $mails = $this->imap('fetch_overview', [implode(',', $mailsIds), (SE_UID == $this->imapSearchOption) ? FT_UID : 0]);
```
Not sure what FT_UID is used for, that why I am a little concerned about writing my own implementation

**Comment by Sebi94nbg** [comment]
@avinash403 
`FT_UID` tells the function, so IMAP, to treat the msg_number argument as UID. Otherwise it's simply an incremental integer. Enabling this, slows the IMAP function usually down.

I fetch all my emails without marking them as "read" and only after processing them I mark them manually as "read". See these examples:

- https://github.com/barbushin/php-imap/wiki/Getting-Started
- https://github.com/barbushin/php-imap/blob/master/examples/get_and_parse_unseen_emails.php
